curriculum/challenges/english/25-front-end-development/quiz-javascript-problem-solving-and-algorithmic-thinking/66edd4bc0ce399f475ad6f9b.md

```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --text--
 
-Placeholder question
+The Fibonacci sequence is a series where each number is the sum of the two preceding ones, starting with `0` and `1`. The sequence goes: `0, 1, 1, 2, 3, 5, 8, 13, 21...` and so on. How can the nth number in this sequence be calculated?
 
 #### --distractors--
 
-Placeholder distractor 1
+Use the formula `fib(n) = fib(n-1) * fib(n-2)`.
 
 ---
 
-Placeholder distractor 2
+Use recursion with memoization to avoid recalculating subproblems where `fib(n) = (n-1) + (n-2)`.
 
 ---
 
-Placeholder distractor 3
+Use a for loop to calculate iteratively where `fib(n) = fib(n-1) + (n-2)`.
 
 #### --answer--
 
-Placeholder answer
+Use recursion with memoization to avoid recalculating subproblems where `fib(n) = fib(n-1) + fib(n-2)`.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+A binary string, such as `"1011"`, represents a number in base-2, where each digit corresponds to a power of `2`. What process can be used to convert this binary string into its decimal equivalent?
 
 #### --distractors--
 
-Placeholder distractor 1
+Multiply each character by powers of `2` and sum them up.
 
 ---
 
-Placeholder distractor 2
+Convert the string to an array, then sum the digits using bitwise operations.
 
 ---
 
-Placeholder distractor 3
+Use the `Number()` function directly with a binary string.
 
 #### --answer--
 
-Placeholder answer
+Use the `parseInt()` function with base `2`.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following algorithm will move all zeroes in an array to the end while keeping the order of other elements unchanged? For example, given `[0, 1, 0, 3, 12]`, return `[1, 3, 12, 0, 0]`.
 
 #### --distractors--
 
-Placeholder distractor 1
+Sort the array in decreasing order and place zeros at the start.
 
 ---
 
-Placeholder distractor 2
+Sort the array in ascending order.
 
 ---
 
-Placeholder distractor 3
+Use recursion to remove all zeros and add them to the end.
```

Comment:
I think this also could work.

Reply:
yeah, forgot to ask for best approach in question. let me update it.